jlm/hls/backend/firrtl2verilog/FirrtlToVerilogConverter.hpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2024 Magnus Sjalander <work@sjalander.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#ifndef JLM_HLS_BACKEND_FIRRTL2VERILOG_FIRRTLTOVERILOGCONVERTER_HPP
+#define JLM_HLS_BACKEND_FIRRTL2VERILOG_FIRRTLTOVERILOGCONVERTER_HPP
+
+#include <jlm/util/file.hpp>
+
+#include <circt/Conversion/ExportVerilog.h>
+#include <circt/Dialect/FIRRTL/FIRParser.h>
+#include <circt/Dialect/HW/HWOps.h>
+#include <circt/Dialect/SV/SVPasses.h>
+#include <circt/Firtool/Firtool.h>
+#include <llvm/Support/SourceMgr.h>
+#include <mlir/IR/BuiltinOps.h>
+#include <mlir/Support/FileUtilities.h>
+
+namespace jlm::hls
+{
+
+using namespace circt;
+using namespace llvm;
+
+/**
+ * Converts FIRRTL to Verilog by reading FIRRTL from a file and writing the converted Verilog to another file.
+ * The functionality is heavily inspired by the processBuffer() function of the firtool in the CIRCT project.
+ *
+ * \param inputFirrtlFile The complete path to the FIRRTL file to convert to Verilog.
+ * \param outputVerilogFile The complete path to the Verilog file to write the converted Verilog to.
+ */
+static void
+FirrtlToVerilogConverter(const std::string inputFirrtlFile, const std::string outputVerilogFile)
+{
+  mlir::MLIRContext context;
+  mlir::TimingScope ts;
+
+  // Set up and read the input FIRRTL file
+  std::string errorMessage;
+  auto input = mlir::openInputFile(inputFirrtlFile, &errorMessage);
+  if (!input) {
+    llvm::errs() << errorMessage << "\n";
+    exit(1);
```

Comment:
I really do not like it that the conversion method directly bails out here. This function should return either an error message or success, and the user should decide what to do on error.

Reply:
I changed it to return a bool